packages/@aws-cdk/aws-synthetics/test/test-class.test.ts

```diff
@@ -0,0 +1,26 @@
+import '@aws-cdk/assert/jest';
+import { App, Stack } from '@aws-cdk/core';
+import * as synthetics from '../lib';
+
+let stack: Stack;
+beforeEach(() => {
+  stack = new Stack(new App(), 'canaries');
+});
+
+test('can specify custom test', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('exports.handler = async () => {\nconsole.log(\'hello world\');\n};'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Code: {
+      Handler: 'index.handler',
+      Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+    },
+  });
+});
```

Comment:
Wasn't this already test with `Canary` or `Code`? I don't understand what this test does here?

Reply:
So from your previous comment, I removed canary references in `Code`. So now, my tests should do as they intend:

- `test-class` tests `test.custom()` (and future static test methods will get tested here)
- `canary` tests other properties of the canary resource
- `code` tests `code.fromInline/fromAsset/fromBucket` in isolation (no canaries)
- `metric` tests metric methods